config/configload/backend.go

```diff
@@ -219,14 +204,25 @@ func newTokenRequestBackend(helper *helper, parent hcl.Body) (map[string]hcl.Bod
 			return nil, err
 		}
 
-		be, berr := PrepareBackend(helper, "", conf.URL, conf)
+		hclbody.RenameAttribute(content, "headers", "set_request_headers")
+		hclbody.RenameAttribute(content, "query_params", "set_query_params")
+		conf.Remain = hclbody.MergeBodies(leftOvers, hclbody.New(content))
+
+		tokenRequestBackend, berr := PrepareBackend(helper, "", conf.URL, conf)
 		if berr != nil {
 			return nil, berr
 		}
-		tokenRequestBackends[label] = be
+
+		if tokenRequestBackend == nil {
+			continue
+		}
+
+		wrapped := wrapBlock(tokenRequest, label,
```

Comment:
For oauth2, the wrapping and merging is still in `PrepareBackend()` and not in `newOAuthBackend()`. Move?